modules/architecture-machine-config-pools.adoc

```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * architecture/control-plane.adoc
+
+[id="architecture-machine-config-pools_{context}"]
+= Node configuration management with machine config pools
+
+Machines that manage the cluster and its workloads are divided into groups based on the types of resources they handle. These groups of machines are called machine config pools (MCP). Each MCP manages a set of nodes and its corresponding machine configs. The role of the node determines which MCP it belongs to; the MCP governs nodes based on its assigned node role label. Nodes in an MCP have the same configuration; this means nodes can be scaled up and torn down in response to increased or decreased workloads.
```

Comment:
Minor: "Machines that manage the cluster" sounds a bit off. Do you think "Machines that run control plane components or user workloads" would be a better framing?